bigquery/v2/apiv2_monoclient/monoclient.go

```diff
@@ -0,0 +1,153 @@
+	}
+	mc.rapClient, err = bigquery.NewRowAccessPolicyClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewRowAccessPolicyClient: %w", err))
+	}
+	mc.tblClient, err = bigquery.NewTableClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewTableClient: %w", err))
+	}
+	if len(errs) > 0 {
+		return nil, fmt.Errorf("NewClient: %d errors during instantiation, first error: %w", len(errs), errs[0])
+	}
+	return mc, nil
+}
+
+// NewClient creates a new monoclient based on REST.
+// The returned client must be Closed when it is done being used to clean up its underlying connections.
+func NewRESTClient(ctx context.Context, opts ...option.ClientOption) (*monoClient, error) {
+	var errs []error
+	var err error
+	mc := &monoClient{}
+	mc.dsClient, err = bigquery.NewDatasetRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewDatasetRESTClient: %w", err))
+	}
+	mc.jobClient, err = bigquery.NewJobRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewJobRESTClient: %w", err))
+	}
+	mc.modelClient, err = bigquery.NewModelRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewModelRESTClient: %w", err))
+	}
+	mc.projClient, err = bigquery.NewProjectRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewProjectRESTClient: %w", err))
+	}
+	mc.routClient, err = bigquery.NewRoutineRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewRoutineRESTClient: %w", err))
+	}
+	mc.rapClient, err = bigquery.NewRowAccessPolicyRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewRowAccessPolicyRESTClient: %w", err))
+	}
+	mc.tblClient, err = bigquery.NewTableRESTClient(ctx, opts...)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("NewTableRESTClient: %w", err))
+	}
+	if len(errs) > 0 {
+		return nil, fmt.Errorf("NewClient: %d errors during instantiation, first error: %w", len(errs), errs[0])
+	}
+	return mc, nil
+}
+
+func (mc *monoClient) Close() error {
+	var errs []error
+	err := mc.dsClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("DatasetClient: %w", err))
+	}
+	err = mc.jobClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("JobClient: %w", err))
+	}
+	err = mc.modelClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("ModelClient: %w", err))
+	}
+	err = mc.projClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("ProjectClient: %w", err))
+	}
+	err = mc.routClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("RoutineClient: %w", err))
+	}
+	err = mc.rapClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("RowAccessPolicyClient: %w", err))
+	}
+	err = mc.tblClient.Close()
+	if err != nil {
+		errs = append(errs, fmt.Errorf("TableClient: %w", err))
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("closing subclients yielded %d errors, first error: %w", len(errs), errs[0])
```

Comment:
might worth adding a `groupErr` function to combine the errors into a single one, and then check for `nil`

```
// [...]
err = groupErr(errs)
if err != nil {
   return fmt.Errorf("err closing subclients: %w", err)
}
// [...]

func groupErr(errs []error) error {
	if len(errs) == 0 {
		return nil
	}
	var errMsgs []string
	for _, err := range errs {
		errMsgs = append(errMsgs, err.Error())
	}
	return fmt.Errorf("found %d errors: %v", len(errs), strings.Join(errMsgs, ", "))
}
```

Reply:
oh, totally forgot about `errors.Join`, good call